widget/text.go

```diff
@@ -359,7 +360,12 @@ func lineBounds(text []rune, wrap fyne.TextWrap, maxWidth int, measurer func([]r
 		switch wrap {
 		case fyne.TextTruncate:
 			for {
-				if measurer(text[low:high]) <= maxWidth {
+				curWidth := measurer(text[low:high])
+				symbWidth := (curWidth / len(text[low:high]))
+				if maxWidth/symbWidth < len(text[low:high]) {
+					high = low + (maxWidth / symbWidth)
```

Comment:
If you change high, shouldn't you re-calculate curWidth?

Reply:
Done